src/entities/friendship_history.rs

```diff
@@ -22,55 +27,88 @@ pub struct FriendshipHistory {
     pub metadata: Option<Json<HashMap<String, String>>>,
 }
 
+// #[automock]
 impl FriendshipHistoryRepository {
     pub fn new(db: Arc<Option<DBConnection>>) -> Self {
         Self { db_connection: db }
     }
 
-    pub async fn create(
+    pub fn create_query<'a>(
         &self,
         friendship_id: Uuid,
-        event: &str,
-        acting_user: &str,
+        event: &'a str,
+        acting_user: &'a str,
         metadata: Option<Json<HashMap<String, String>>>,
-    ) -> Result<(), sqlx::Error> {
-        let db_conn = DatabaseComponent::get_connection(&self.db_connection);
-        match sqlx::query(
+    ) -> Query<'a, Postgres, sqlx::postgres::PgArguments> {
+        sqlx::query(
                 "INSERT INTO friendship_history (id,friendship_id, event, acting_user, metadata) VALUES ($1,$2,$3,$4,$5)",
         )
         .bind(Uuid::parse_str(generate_uuid_v4().as_str()).unwrap())
         .bind(friendship_id)
         .bind(event)
         .bind(acting_user)
         .bind(metadata)
-        .execute(db_conn)
-        .await
-        {
-            Ok(_) => Ok(()),
-            Err(err) => Err(err),
+    }
+
+    pub async fn create<'a>(
+        &'a self,
+        friendship_id: Uuid,
+        event: &'a str,
+        acting_user: &'a str,
+        metadata: Option<Json<HashMap<String, String>>>,
+        transaction: Option<Transaction<'a, Postgres>>,
+    ) -> (Result<(), sqlx::Error>, Option<Transaction<'a, Postgres>>) {
+        let executor = self.get_executor(transaction);
+
+        let query = self.create_query(friendship_id, event, acting_user, metadata);
+
+        let (res, resulting_executor) = DatabaseComponent::execute_query(query, executor).await;
+
+        let transaction_to_return = get_transaction_result_from_executor(resulting_executor);
+
+        match res {
+            Ok(_) => (Ok(()), transaction_to_return),
+            Err(err) => (Err(err), transaction_to_return),
         }
```

Comment:
I think this might be a simpler expression of the same.
```suggestion
        (res.map(|_| ()), transaction_to_return)
```